app/models/season/phase_switcher.rb

```diff
@@ -4,6 +4,17 @@ def self.season
     Season.current
   end
 
+  def self.destined(phase)
+    unless phase.to_sym.in?(whitelisted_phases)
+      raise ArgumentError
+    else
+      self.public_send(phase)
+    end
+  end
+
```

Comment:
It's more the `unless` condition: converting a String to Symbal and iterating over an Array. Should that take 4ms? I don't know. It's not that we're running at webscale here so it doesn't really matter.

As for the raise: yeah, maybe the silent fail is indeed enough. You'd still need the whitelist check, but don't need to raise anything.